examples/hal/quad/main.rs

```diff
@@ -617,3 +598,218 @@ fn main() {
+        <back::Backend as hal::Backend>::ImageView,
+    )>,
+    <back::Backend as hal::Backend>::GraphicsPipeline,
+    <back::Backend as hal::Backend>::PipelineLayout,
+) {
+    let (pixel_width, pixel_height) =
+        surface.update_dimensions(pixel_width as _, pixel_height as _, &physical_device);
+    println!("Surface format: {:?}", surface_format);
+    let swap_config = SwapchainConfig::new()
+        .with_color(*surface_format)
+        .with_image_usage(i::Usage::COLOR_ATTACHMENT);
+    let (swap_chain, backbuffer) = device.create_swapchain(surface, swap_config, None);
+
+    let render_pass = {
+        let attachment = pass::Attachment {
+            format: Some(*surface_format),
+            samples: 1,
+            ops: pass::AttachmentOps::new(
+                pass::AttachmentLoadOp::Clear,
+                pass::AttachmentStoreOp::Store,
+            ),
+            stencil_ops: pass::AttachmentOps::DONT_CARE,
+            layouts: i::Layout::Undefined..i::Layout::Present,
+        };
+
+        let subpass = pass::SubpassDesc {
+            colors: &[(0, i::Layout::ColorAttachmentOptimal)],
+            depth_stencil: None,
+            inputs: &[],
+            resolves: &[],
+            preserves: &[],
+        };
+
+        let dependency = pass::SubpassDependency {
+            passes: pass::SubpassRef::External..pass::SubpassRef::Pass(0),
+            stages: PipelineStage::COLOR_ATTACHMENT_OUTPUT..PipelineStage::COLOR_ATTACHMENT_OUTPUT,
+            accesses: i::Access::empty()
+                ..(i::Access::COLOR_ATTACHMENT_READ | i::Access::COLOR_ATTACHMENT_WRITE),
+        };
+
+        device.create_render_pass(&[attachment], &[subpass], &[dependency])
+    };
+    let (frame_images, framebuffers) = match backbuffer {
+        Backbuffer::Images(images) => {
+            let extent = i::Extent {
+                width: pixel_width as _,
+                height: pixel_height as _,
+                depth: 1,
+            };
+            let pairs = images
+                .into_iter()
+                .map(|image| {
+                    let rtv = device
+                        .create_image_view(
+                            &image,
+                            i::ViewKind::D2,
+                            *surface_format,
+                            Swizzle::NO,
+                            COLOR_RANGE.clone(),
+                        )
+                        .unwrap();
+                    (image, rtv)
+                })
+                .collect::<Vec<_>>();
+            let fbos = pairs
+                .iter()
+                .map(|&(_, ref rtv)| {
+                    device
+                        .create_framebuffer(&render_pass, Some(rtv), extent)
+                        .unwrap()
+                })
+                .collect();
+            (pairs, fbos)
+        }
+        Backbuffer::Framebuffer(fbo) => (Vec::new(), vec![fbo]),
+    };
+
+    let pipeline_layout =
```

Comment:
The render pass needs to be recreated because it depends on the format of the swap chain images, which can, however unlikely, change.. The pipeline depends on that renderpass.

Reply:
Interesting. Aren't *we* dictating the format of swapchain images? Why/when would it change?